### ISSUE TITLE
Can we turn Mask_RCNN in a package?

### ISSUE THREAD

**Issue by jerpint** [matterport/Mask_RCNN#485]
One thing I've noticed from using Mask_RCNN is that it is designed not a package completely. This means that suppose you want to use `Mask_RCNN` as a building block in code, it will break. For example:

Suppose I have a repo, `MyProject/` and contained within it I have a forked version of `Mask_RCNN` as such:
├── MyProject
├── some_code.py
├── Mask_RCNN
│   ├── assets
│   ├── coco.py
│   ├── ...

If I want to use any of the useful functions in `some_code.py` , I would need to run 

`from Mask_RCNN import coco` . 

This first off requires `Mask_RCNN/` to contain an `__init__.py`, and second, in importing `coco.py`, the current statement is `import utils` which should then at this point be imported not directly, but from Mask_RCNN.  There are similar examples in `visualize.py` and `model.py`.

I can create a pull request for this, it only requires a couple of minor changes in the headers of

* `coco.py`
* `visualize.py*
* `model.py*

**Comment by CMCDragonkai** [comment]
In my situation:

```
solar-panel-analysis/
  Mask_RCNN/
  solar/
    __init__.py
    main.py
  script.py
```

Inside `script.py`:

```py
from Mask_RCNN.mrcnn import utils
```

Works.

However inside `main.py`. Doing the same thing doesn't work. It cannot find it. I think this requires doing the `python setup.py install` inside the `Mask_RCNN` which will make `mrcnn` a global package.

**Comment by CMCDragonkai** [comment]
You don't need to have `__init__.py` for python 3.3+. I'm running 3.6 so that's why it works.

**Comment by CMCDragonkai** [comment]
Hmm I think I have encountered a problem you are talking about.

The `model.py` performs `from mrcnn import utils`, and if `Mask_RCNN` repository is in that directory structure stated in your OP, then importing `from Mask_RCNN.mrcnn import model as modellib` fails with:

```
ImportError: No module named 'mrcnn'
```

**Comment by CMCDragonkai** [comment]
Swapping `from mrcnn import utils` to `from Mask_RCNN.mrcnn import utils` does end up working.

**Comment by eyildiz-ugoe** [comment]
@CMCDragonkai  Not really, you just get another error which says `ModuleNotFoundError: No module named 'Mask_RCNN'`.

**Comment by MichaelTendoSsemwanga** [comment]
I am trying to run the notebooks, demo.ipynb in colab but, nothing imports correct

**Comment by JulienMaille** [comment]
> I am trying to run the notebooks, demo.ipynb in colab but, nothing imports correct

I have the same issue, did you found a solution?
